031-custom_authentication.md

```diff
@@ -0,0 +1,104 @@
+# Custom Authentication
+
+This proposal focuses on supporting custom authentication in the Strimzi operator.
+
+## Current situation
+
+Currently, numerous authentication methods are [supported](https://github.com/strimzi/strimzi-kafka-operator/tree/0.25.0/api/src/main/java/io/strimzi/api/kafka/model/authentication) in Strimzi, and allows the user to configure which one to use. In addition, in the case a custom *authorizer* needs to be used, this is also currently [supported](https://github.com/strimzi/strimzi-kafka-operator/blob/0.25.0/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java#L541-L549). 
+
+However, there is no ability to specify custom authentication, which is what this proposal will focus on.
+
+## Motivation
+
+Supporting custom authentication allows greater flexibility for users of Strimzi who may have bespoke/proprietary authentication requirements. This would greatly broaden user adoption of the operator, as it would also support more common authentication schemes, such as [Amazon MSK](https://docs.aws.amazon.com/msk/latest/developerguide/security_iam_service-with-iam.html), and attract users who have much more stringent requirements (i.e. proprietary auth). 
+
+The “cost” of supporting such feature should be minimal as well, as the expectation for custom authentication passes the complexity onto the user. Strimzi should strictly support the passing of configuration values, and nothing else. It’s up to the user to ensure their authn/z schemes can be integrated with Strimzi. Considering this also requires custom built Strimzi Kafka images in order to load custom classes on to the classpath, it’s also expected that will only really apply to highly technical/pro users.
+
+## Proposal
+
+Add a CustomAuthenticationKafkaListener class, which supports an example of the following properties
+
+```
```

Comment:
> Sorry, I was away for some time. I left some more comment. These custom things are also very hard to test. So it would be good to also include some strategy for system tests covering this feature. Since without that it would be very hard to maintain this code.

No worries whatsoever! Agreed, I'll include a testing plan as well.